librz/include/rz_th.h

```diff
@@ -55,6 +50,12 @@ RZ_API int rz_th_lock_tryenter(RzThreadLock *thl);
 RZ_API int rz_th_lock_enter(RzThreadLock *thl);
 RZ_API int rz_th_lock_leave(RzThreadLock *thl);
 RZ_API void *rz_th_lock_free(RzThreadLock *thl);
+#define rz_th_lock_guard(lock, x) \
+	do { \
+		rz_th_lock_enter(lock); \
+		x; \
+		rz_th_lock_leave(lock); \
+	} while (0)
```

Comment:
I don't think this is a good idea. If you e.g. `return` or `break` inside of it, it will keep the lock.

Reply:
yeah i thought about this... there is no easy way to do this in C unfortunately